File: spring-webflux-video-streaming-example/src/main/java/com/knf/demo/config/EndPointConfig.java
package com.knf.demo.config;
import com.knf.demo.service.VideoStreamingService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.io.Resource;
import org.springframework.http.MediaType;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.RouterFunctions;
import org.springframework.web.reactive.function.server.ServerRequest;
import org.springframework.web.reactive.function.server.ServerResponse;
import reactor.core.publisher.Mono;

@Configuration
public class EndPointConfig {
    @Autowired
    private VideoStreamingService service;

    @Bean
    public RouterFunction<ServerResponse> router(){
        return RouterFunctions.route()
                .GET("video/{name}", this::videoHandler)
                .build();
    }

    private Mono<ServerResponse> videoHandler(ServerRequest serverRequest){
        String title = serverRequest.pathVariable("name");
        return ServerResponse.ok()
                .contentType(MediaType.valueOf("video/mp4"))
                .body(this.service.getVideo(title), Resource.class);
    }

}

File: kotlin-spring-main/springkotlinwebfluxvideostreaming/src/main/kotlin/com/knf/springkotlinwebfluxvideostreaming/config/EndPointConfig.kt
package com.knf.springkotlinwebfluxvideostreaming.config


import com.knf.springkotlinwebfluxvideostreaming.service.VideoStreamingService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.io.Resource
import org.springframework.http.MediaType
import org.springframework.web.reactive.function.server.*
import reactor.core.publisher.Mono

@Configuration
class EndPointConfig {
    @Autowired
    private val service: VideoStreamingService? = null
    @Bean
    fun router(): RouterFunction<ServerResponse> {
        return RouterFunctions.route()
            .GET(
                "video/{name}",
                HandlerFunction<ServerResponse> { serverRequest: ServerRequest -> videoHandler(serverRequest) })
            .build()
    }

    private fun videoHandler(serverRequest: ServerRequest): Mono<ServerResponse> {
        val title: String = serverRequest.pathVariable("name")
        return ServerResponse.ok()
            .contentType(MediaType.valueOf("video/mp4"))
            .body(service!!.getVideo(title), Resource::class.java)
    }
}